roles/letsencrypt/templates/nginx-challenge-site.conf.j2

```diff
@@ -1,5 +1,5 @@
 server {
   listen 80;
-  server_name {{ item.item.value.site_hosts | reverse_www(enabled=item.item.value.www_redirect | default(true)) | join(' ') }};
+  server_name{% for item in item.item.value.site_hosts %} {{ item.canonical }}{% for redirect in item.redirects | default([]) %} {{ redirect }}{% endfor %}{% endfor %};
```

Comment:
The disappointment here is that it uses `item.item.value.site_hosts`.
We would need just `item.value.site_hosts` in order to use `site_hosts` from `helpers.yml`.

I tried opening the template with `{% set item = item.item %}` but it doesn't work.
If it feels cleaner or more consistent, we could open the template with
`{% set canonical = ... [something] %}`
`{% set redirects = ... [something] %}`
`{% set site_hosts = ... [something with union] %}`
and _then_ do the `site_hosts | join(' ')`